### PR TITLE
CCM-2422

### PR DESCRIPTION
**Pull request by PhilipSkinner**
## Summary

Adds new GET /v1/messages/{messageId} endpoint documentation.

Refactors small bits of docs, fixes some minor bugs in other endpoint specifications.

## Reviews Required
* [x] Dev
* [ ] Test
* [ ] Tech Author
* [ ] Product Owner

## Checklist
* [ ] Brief description of work completed, and any technical decisions made as part of the PR
* [ ] PR link added as a comment to the relevant JIRA ticket
* [ ] Branch deployed to a dynamic env - link to deployment pipeline
* [ ] PR link shared on Slack and/or Teams
* [ ] 1 reviews received

### CODE REVIEW

File: sandbox/messages/2bBBpsiMl2rnQt99qm6JLZ6w1vq.json [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+          "routingPlan" : {
+            "id" : "b838b13c-f98c-4def-93f0-515d4e4f4ee1-override",
+            "type" : "override",
+            "version" : "5"
+          }
+        }
+      ],
+      "timestamps": {
+        "created": "2023-10-09T10:31:37Z",
+        "updated": "2023-10-09T10:52:12Z",
```
Comment (PhilipSkinner) [inline sandbox/messages/2bBBpsiMl2rnQt99qm6JLZ6w1vq.json:33]: Updated timestamp needs to be removed, should not be present on API responses

File: specification/schemas/responses/MessageResponse.yaml [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+properties:
+  data:
+    type: object
+    additionalProperties: false
+    properties:
+      type:
+        $ref: ../enums/TypeMessage.yaml
+      id:
+        $ref: ../types/KSUID.yaml
+        description: Identifier for this Message.
```
Comment (PhilipSkinner) [inline specification/schemas/responses/MessageResponse.yaml:13]: Missing the example on this, needs to be the same value as in the `self` url below: `0ujsszwN8NRY24YaXiTIE2VWDTS `

File: specification/schemas/responses/MessageCreated.yaml [hunk 3]
```diff
@@ -16,10 +16,7 @@ properties:
...
         properties:
           messageReference:
-            type: string
-            description: Your unique message reference.
-            format: uuid
-            example: da0b1495-c7cb-468c-9d81-07dee089d728
+            $ref: ../types/MessageReference.yaml
           messageStatus:
             description: The status of your message.
             $ref: ../enums/MessageCreatedStatus.yaml
```
Comment (PhilipSkinner) [inline specification/schemas/responses/MessageCreated.yaml:22]: I believe we're missing the `links` section on this response

Comment (TheWorldWideWebb) [reply specification/schemas/responses/MessageCreated.yaml:22]: added as a template and replaced on both response and request

File: specification/schemas/responses/MessageResponse.yaml (hunk 2 above)
Comment (TheWorldWideWebb) [reply specification/schemas/responses/MessageResponse.yaml:13]: Added example

File: sandbox/messages/2bBBpsiMl2rnQt99qm6JLZ6w1vq.json [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        "enriched": "2023-10-09T10:31:59Z",
+        "failed": "2023-10-09T10:52:12Z"
+      },
+      "metadata": [
+        {
+          "queriedAt": "2023-10-09T10:31:59Z",
+          "versionId": "23",
+          "source": "pds"
```
[the remark this answers is not in the document]
Comment (TheWorldWideWebb) [reply sandbox/messages/2bBBpsiMl2rnQt99qm6JLZ6w1vq.json:41]: added sms in the label array

File: sandbox/messages/2bBBpsiMl2rnQt99qm6JLZ6w1vq.json (hunk 1 above)
Comment (TheWorldWideWebb) [reply sandbox/messages/2bBBpsiMl2rnQt99qm6JLZ6w1vq.json:33]: removed

File: sandbox/messages/2bBBpsiMl2rnQt99qm6JLZ6w1vq.json [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+          "retryCount": 0,
+          "channelStatus": "delivered",
+          "channelStatusDescription": "Contact detail is missing",
+          "timestamps": {
+            "created": "2023-10-09T10:31:59Z",
+            "updated": "2023-10-09T10:52:12Z",
+            "failed": "2023-10-09T10:52:12Z"
+          },
+          "routingPlan" : {
+            "id" : "b838b13c-f98c-4def-93f0-515d4e4f4ee1-override",
```
[the remark this answers is not in the document]
Comment (TheWorldWideWebb) [reply sandbox/messages/2bBBpsiMl2rnQt99qm6JLZ6w1vq.json:25]: guid now used

File: sandbox/messages/2bBBpsiMl2rnQt99qm6JLZ6w1vq.json [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+      ],
+      "timestamps": {
+        "created": "2023-10-09T10:31:37Z",
+        "enriched": "2023-10-09T10:31:59Z",
+        "failed": "2023-10-09T10:52:12Z"
+      },
+      "metadata": [
+        {
+          "queriedAt": "2023-10-09T10:31:59Z",
+          "versionId": "23",
```
Comment (PhilipSkinner) [inline sandbox/messages/2bBBpsiMl2rnQt99qm6JLZ6w1vq.json:39]: Should be `version` not `versionId` - wondering where you copied this from as all of the example messages should have been updated to be compatible with the spec? Can we make sure we are rebased on `release` please?

File: sandbox/messages/2bBBpsiMl2rnQt99qm6JLZ6w1vq.json [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+      "messageStatusDescription": "Contact detail is missing",
+      "channels": [
+        {
+          "type": "sms",
+          "retryCount": 0,
+          "channelStatus": "delivered",
+          "channelStatusDescription": "Contact detail is missing",
+          "timestamps": {
+            "created": "2023-10-09T10:31:59Z",
+            "updated": "2023-10-09T10:52:12Z",
```
Comment (PhilipSkinner) [inline sandbox/messages/2bBBpsiMl2rnQt99qm6JLZ6w1vq.json:21]: `updated` timestamp shouldn't be present - this looks like it was copied from an old example message - see previous comment about making sure we're rebased on `release`

File: sandbox/messages/2bBBpsiMl2rnQt99qm6JLZ6w1vq.json [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+    "type": "Message",
+    "id": "2bBBpsiMl2rnQt99qm6JLZ6w1vq",
+    "attributes": {
+      "routingPlan": {
+        "id" : "b838b13c-f98c-4def-93f0-515d4e4f4ee1",
+        "version" : "1"
+      },
+      "messageReference": "1226417c-f683-4d68-afe8-2d52321a6b70",
+      "messageStatus": "delivered",
+      "messageStatusDescription": "Contact detail is missing",
```
Comment (PhilipSkinner) [inline sandbox/messages/2bBBpsiMl2rnQt99qm6JLZ6w1vq.json:12]: Should be `Failed reason: Contact detail is missing`

File: sandbox/messages/2bBBpsiMl2rnQt99qm6JLZ6w1vq.json [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+      "channels": [
+        {
+          "type": "sms",
+          "retryCount": 0,
+          "channelStatus": "delivered",
+          "channelStatusDescription": "Contact detail is missing",
```
Comment (PhilipSkinner) [inline sandbox/messages/2bBBpsiMl2rnQt99qm6JLZ6w1vq.json:18]: Should be `Failed reason: Contact detail is missing`

File: sandbox/messages/2bBBpsiMl2rnQt99qm6JLZ6w1vq.json [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+        "version" : "1"
+      },
+      "messageReference": "1226417c-f683-4d68-afe8-2d52321a6b70",
+      "messageStatus": "delivered",
+      "messageStatusDescription": "Contact detail is missing",
+      "channels": [
+        {
+          "type": "sms",
+          "retryCount": 0,
+          "channelStatus": "delivered",
```
Comment (PhilipSkinner) [inline sandbox/messages/2bBBpsiMl2rnQt99qm6JLZ6w1vq.json:17]: should be `failed`

File: sandbox/messages/2bBBpsiMl2rnQt99qm6JLZ6w1vq.json [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+            "id" : "9ba00d23-cd6f-4aca-8688-00abc85a7980",
+            "type" : "override",
+            "version" : "5"
+          }
+        }
+      ],
+      "timestamps": {
+        "created": "2023-10-09T10:31:37Z",
+        "enriched": "2023-10-09T10:31:59Z",
+        "failed": "2023-10-09T10:52:12Z"
```
Comment (PhilipSkinner) [inline sandbox/messages/2bBBpsiMl2rnQt99qm6JLZ6w1vq.json:34]: in this scenario the failed time should be the same as the enriched time - this message failed due to PDS data missing the patients mobile telephone number - can we get the `failed` timestamp updated to have the same value as the `enriched` timestamp?

File: specification/schemas/responses/MessageCreated.yaml (hunk 3 above)
Comment (PhilipSkinner) [reply specification/schemas/responses/MessageCreated.yaml:22]: that'll do for now! when we add new resources we'll need to rename/refactor this bit to be more specific

File: specification/schemas/responses/MessageCreated.yaml [hunk 12]
```diff
@@ -16,10 +16,7 @@ properties:
         additionalProperties: false
```
Comment (PhilipSkinner) [inline specification/schemas/responses/MessageCreated.yaml:16]: Just noticed that we are missing the `Location` header on the POST /v1/messages response!

File: specification/schemas/responses/MessageCreated.yaml [hunk 13]
```diff
@@ -35,3 +32,5 @@ properties:
           routingPlan:
             description: The routing plan that you requested the message be sent with.
             $ref: ../types/RoutingPlan.yaml
+        links:
```
Comment (PhilipSkinner) [inline specification/schemas/responses/MessageCreated.yaml:35]: This appears to be in the wrong place in the document, its not coming through when I generate and preview the docs:

image

File: specification/schemas/types/Links.yaml [hunk 14]
```diff
@@ -0,0 +1,9 @@
+type: object
```
Comment (PhilipSkinner) [inline specification/schemas/types/Links.yaml:1]: I think this has broken the documentation on the GET /v1/messages/{messageId} endpoint somehow:

image

Its now showing as an `any` with no properties defined on it

File: specification/schemas/responses/MessageCreated.yaml (hunk 13 above)
Comment (TheWorldWideWebb) [reply specification/schemas/responses/MessageCreated.yaml:35]: missing $ on ref tags

**Review by TheWorldWideWebb — APPROVED** [review]

**Pull request merged**